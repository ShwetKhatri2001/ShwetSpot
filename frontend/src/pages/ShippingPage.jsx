import React, { useState } from 'react';
import { Button, Form } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import CheckoutBar from '../components/CheckoutBar';
import { saveShippingAddress } from '../redux/actions/cartActions';

const ShippingPage = ({ history }) => {
  const cart = useSelector(state => state.cart);
  const { shippingAddress } = cart;

  const [address, setAddress] = useState(shippingAddress.address);
  const [city, setCity] = useState(shippingAddress.city);
  const [postalCode, setPostalCode] = useState(shippingAddress.postalCode);
  const [country, setCountry] = useState(shippingAddress.country);

  const dispatch = useDispatch();

  const submitHandler = e => {
    e.preventDefault();
    dispatch(saveShippingAddress({ address, city, postalCode, country }));
    history.push('/payment');
  };

  return (
    <>
      <CheckoutBar step1 step2 />
      <div className="justify-content-center align-items-center container bg-light p-3">
        <h1 className="pl-3">Shipping</h1>
        <Form onSubmit={submitHandler} className="pr-3 pl-3">
          <Form.Group>
            <Form.Label>Address</Form.Label>
            <Form.Control
              type="text"
              placeholder="Enter address"
              value={address}
              required
              onChange={e => setAddress(e.target.value)}
            />
          </Form.Group>

          <Form.Group>
            <Form.Label>City</Form.Label>
            <Form.Control
              type="text"
              placeholder="Enter city"
              value={city}
              required
              onChange={e => setCity(e.target.value)}
            />
          </Form.Group>

          <Form.Group>
            <Form.Label>Postal Code</Form.Label>
            <Form.Control
              type="text"
              placeholder="Enter postal code"
              value={postalCode}
              required
              onChange={e => setPostalCode(e.target.value)}
            />
          </Form.Group>

          <Form.Group>
            <Form.Label>Country</Form.Label>
            <Form.Control
              type="text"
              placeholder="Enter country"
              value={country}
              required
              onChange={e => setCountry(e.target.value)}
            />
          </Form.Group>

          <Button type="submit" variant="success" className="mb-2">
            Continue
          </Button>
        </Form>
      </div>
    </>
  );
};

export default ShippingPage;
