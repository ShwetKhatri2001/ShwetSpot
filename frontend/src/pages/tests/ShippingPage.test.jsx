import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import ShippingPage from '../ShippingPage';
import store from '../../redux/store';

describe('ShippingPage Component', () => {
  const MockShippingPage = (
    <MemoryRouter>
      <Provider store={store}>
        <ShippingPage />
      </Provider>
    </MemoryRouter>
  );

  it('should match the snapshot', () => {
    const tree = renderer.create(MockShippingPage).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
