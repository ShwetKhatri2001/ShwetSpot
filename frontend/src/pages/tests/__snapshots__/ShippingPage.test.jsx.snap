// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShippingPage Component should match the snapshot 1`] = `
Array [
  <div
    className="justify-content-center mb-4 nav"
    onKeyDown={[Function]}
    style={
      Object {
        "backgroundColor": "#f8f8f8",
      }
    }
  >
    <div
      className="nav-item"
    >
      <a
        className="nav-link active"
        data-rb-event-key="/login"
        href="/login"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Sign In
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        className="nav-link"
        data-rb-event-key="/shipping"
        href="/shipping"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Shipping
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        aria-disabled={true}
        className="nav-link disabled"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={-1}
      >
        Payment
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        aria-disabled={true}
        className="nav-link disabled"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={-1}
      >
        Place Order
      </a>
    </div>
  </div>,
  <div
    className="justify-content-center align-items-center container bg-light p-3"
  >
    <h1
      className="pl-3"
    >
      Shipping
    </h1>
    <form
      className="pr-3 pl-3"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Address
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter address"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          City
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter city"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Postal Code
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter postal code"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          Country
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter country"
          required={true}
          type="text"
        />
      </div>
      <button
        className="mb-2 btn btn-success"
        disabled={false}
        type="submit"
      >
        Continue
      </button>
    </form>
  </div>,
]
`;
